entity-management-web/src/main/java/eu/europeana/entitymanagement/web/BaseRest.java

```diff
@@ -136,5 +147,35 @@ public String computeEtag(long timestamp, String format, String version){
         return DigestUtils.md5Hex(String.format("%s:%s:%s", timestamp, format, version));
     }
 
+    @SuppressWarnings("unchecked")
+	private void processLanguage(Entity entity, String languages) {
+		if(languages==null) return;
+		List<String> languagesList = Arrays.asList(languages.split(",", -1));
+    	List<Field> entityFields = EntityUtils.getAllFields(entity.getClass());
+		for(Field field : entityFields) {
+			if(field.getType().isAssignableFrom(Map.class)) {
```

Comment:
notation is a map but not multilingual. 
Use EntityFieldsTypes.isMultilingual() instead, but also check if the field value is not null before checking the type

Reply:
Improved.